tests/functional/models/test_device.py

```diff
@@ -256,124 +233,112 @@ def test_move(self):
         with self.assertRaises(self.helper.balena_exceptions.IncompatibleApplication):
             self.balena.models.device.move(device['uuid'], 'FooBarBarBar')
 
-    def test_set_custom_location(self):
-        app, device = self.helper.create_device()
+    def test_20_set_custom_location(self):
         location = {
             'latitude': '41.383333',
             'longitude': '2.183333'
         }
 
         # should be able to set the location of a device by uuid.
-        self.balena.models.device.set_custom_location(device['uuid'], location)
-        device = self.balena.models.device.get(device['uuid'])
-        self.assertEqual(device['custom_latitude'], '41.383333')
-        self.assertEqual(device['custom_longitude'], '2.183333')
+        self.balena.models.device.set_custom_location(type(self).device['uuid'], location)
+        device = self.balena.models.device.get(type(self).device['uuid'])
+        self.assertEqual(type(self).device['custom_latitude'], '41.383333')
```

Comment:
hey @thgreasi this test fails because what it is actually doing is updating a property on the device itself. when you do `self.balena.models.device.set_custom_location(type(self).device['uuid'], location)` this changes the custom location, because now `type(self).device)` points to the old variable, this breaks. 

I think the main problem here was when copying type(self).device over device you overwrote the device on line 244. I think that changing in test 20 and 21 the assert to

```suggestion
self.assertEqual(device['custom_latitude'], '41.383333')
self.assertEqual(device['custom_longitude'], '2.183333')
```

Should fix it

Reply:
Right @otaviojacobi ... I guess that was a signal that I was indeed tired XD